torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2503,7 +2503,8 @@ void XLANativeFunctions::_propagate_xla_data(const at::Tensor& input,
   XLATensor::ShardingSpecPtr sharding = input_tensor->sharding_spec();
   if (sharding && sharding->sharding.type() != xla::OpSharding::UNKNOWN) {
     tensor_methods::custom_sharding_(output_tensor,
-                                     input_tensor->sharding_spec());
+                                     input_tensor->sharding_spec(),
+                                     CustomSharding::Type::kSharding);
```

Comment:
so you assume only tensor with `kSharding` will be called with in place ops?

Reply:
That's the original design which is to align with the original design of SPMD... So yea.. for kSharding...